### DISCUSSION TITLE
Can't add or read tasks after login.

### DISCUSSION THREAD

**Discussion by p7gg** [supabase/supabase#3729] [Questions]
Hey yall, I'm using Angular to basically clone the Microsoft To do App, and I'm having some problems that i don't know exactly why. my issue:

- Whenever I Login into my application I can't add or read any tasks, it gives my this error:
```
{
   code: "42501"
   details: null
   hint: null
   message: "new row violates row-level security policy for table \"tasks\""
}
```

But it works just fine if I refresh the page after I've logged in.

Thats the policy I have setup for the table: `(uid() = user_id)`

I'm happy to provide more details if anyone needs.

**Comment by silentworks** [comment]
It's best to provide all the details you can as the information here is not enough to work out what could be going wrong, It would be good to know what your logging code looks like? I'm guessing its something to do with server/client as I think Angular does SSR.

**Reply by p7gg** [reply]
Sorry, I'm quite new to development and often I struggle to know where I'm going wrong about something...
As I don't know what is really necessary for you to find out, i'll just paste some lines of my code:

auth.service.ts:
```
signIn(email: string, password: string) {
    return this.supabase.auth.signIn({ email, password });
} 
```

login.component.ts:
```
async handleLogin(type: string): Promise<void> {
    const { email, password } = this.authForm.value;

    const { user, error, session } =
      type === 'LOGIN'
        ? await this._authService.signIn(email, password)
        : await this._authService.signUp(email, password);

    if (user && type === 'LOGIN') {
      await this._router.navigate(['/']);
    } else if (error) {
      console.error(error);
      this._snackBar.open(error.message, 'OK');
    } else if (user && !session && !error) {
      this._snackBar.open('A verification has been sent!', 'OK');
    }
  }
```

Then I have a separate service for all the CRUD operations:

supabase.service.ts:
```
async fetchTasks() {
    try {
      const { data, error } = await this.supabase
        .from<Task>('tasks')
        .select('*, steps(*)')

      if (error) {
        throw error;
      }

      this._tasks.next(<Task[]>data);
      console.log('🚀 ~ fetched data: ', data);

      for (let i = 0; i < this._tasks.value.length; i++) {
        this._steps.next([...this._steps.value, ...this._tasks.value[i].steps]);
      }
    } catch (error) {
      console.error(error);
    }
  }
```

Then I call this fetchTasks() method in the tasks.component.ts:
```
async ngOnInit() {

    await this._supabase.fetchTasks();

    this.tasksSubscription = this._supabase.tasks$.subscribe((data) => {
      this.tasks = data;

      const views: { [view: string]: number } = {};

      for (let task of data) {
        for (let view of task.views) {
          views[view] ??= 0;
          views[view] += 1;
        }
      }

      this.views = views;
    });
  }
```

**Reply by p7gg** [reply]
Please, lmk if you need anthing else.
